### DISCUSSION TITLE
Share Prisma client/schema among multiple projects

### DISCUSSION THREAD

**Discussion by niccolofant** [prisma/prisma#14926] [Q&A] [answered]
The question speaks by itself, i'd like to know what's the best strategy to share a Prisma client/schema between different projects in order to have a single point to access the database and don't have duplicated code.

At the moment my backend architecture is as follows:
 - GraphQL server with Prisma configured + DB connection
 - a cloud function which will also need to connect to the same DB 
The problem is, how can share Prisma with the cloud functions without repeating code?

I also tried to create a private npm package with the prisma configuration, following [this tutorial](https://isidoro-ferreiro.medium.com/share-your-prisma-client-across-projects-44d1c7aca6fd) but it doesn't seem the correct solution (maybe i'm wrong tho).

**edit**: Or maybe should i just not use Prisma and make a raw query? 

Thanks

**Comment by austincrim** [comment, marked as answer]
Hey @ocintnaf 👋,

If you are not using a monorepo, a private package is typically the way to go. Were you running into any specific issues with that approach?

**Reply by vancerbtw** [reply]
I am wondering how to handle different package/schema versions for production and dev environments

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
